fdbserver/RestoreCommon.actor.h

```diff
@@ -212,26 +214,31 @@ struct RestoreFileFR {
 		r.endVersion = t.getInt(i++);
 		r.beginVersion = t.getInt(i++);
 		r.cursor = t.getInt(i++);
+		r.fileIndex = t.getInt(i++);
 		return r;
 	}
 
-	bool operator<(const RestoreFileFR& rhs) const { return beginVersion < rhs.beginVersion; }
+	bool operator<(const RestoreFileFR& rhs) const {
+		return beginVersion < rhs.beginVersion || (beginVersion == rhs.beginVersion && endVersion < rhs.endVersion) ||
+		       (beginVersion == rhs.beginVersion && endVersion == rhs.endVersion && fileIndex < rhs.fileIndex);
```

Comment:
Tuples make it easy to write non-trivial comparison operators:

`return std::tie(beginVersion, endVersion, fileIndex) < std::tie(rhs.beginVersion, rhs.endVersion, rhs.fileIndex);`